hikari/internal/cache.py

```diff
@@ -287,16 +296,13 @@ class BaseData(abc.ABC, typing.Generic[ValueT]):
     __slots__: typing.Sequence[str] = ()
 
     @abc.abstractmethod
-    def build_entity(self, **kwargs: typing.Any) -> ValueT:
+    def build_entity(self, app: traits.RESTAware, /) -> ValueT:
         """Build an entity object from this data object.
 
         Parameters
         ----------
-        target
-            The class of the entity object to build.
-        kwargs
-            Extra fields to pass on to the entity's initialiser. These will take
-            priority over fields on the builder.
+        app : hikari.traits.RESTAware
```

Comment:
```suggestion
        app : hikari.traits.BotAware
```

Reply:
If these needed to be BotAware wouldn't mypy have complained, this is only for building the models which are all currently annotated as having RESTAware rather than BotAware

Reply:
They dont need to, but they will be called from the cache, which is BotAware. Its just for consistency sake

Reply:
Originally the cache was typed as taking not aware so it could be intent aware, not really linked to mode handling

Reply:
Ignore this then